hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/publishing/handlers/HandlersChainFactory.java

```diff
@@ -51,7 +62,35 @@ public HttpHandler provide() {
                                                                 messageErrorProcessor, throughputLimiter);
         TopicHandler topicHandler = new TopicHandler(readHandler, topicsCache, messageErrorProcessor);
 
-        return topicHandler;
+        boolean authenticationEnabled = configFactory.getBooleanProperty(FRONTEND_AUTHENTICATION_ENABLED);
+        return authenticationEnabled ? withAuthenticationHandlersChain(topicHandler) : topicHandler;
+    }
+
+    private HttpHandler withAuthenticationHandlersChain(HttpHandler next) {
+        AuthenticationConfiguration authConfig = authenticationConfigurationProvider.getAuthenticationConfiguration()
+                .orElseThrow(() -> new IllegalStateException("AuthenticationConfiguration was not provided"));
+
+        try {
+            return createAuthenticationHandlersChain(next, authConfig);
+        } catch (Exception e) {
+            throw new IllegalStateException("Could not create authentication handlers chain", e);
```

Comment:
@wojtkiewicz is working on the authorisation part of this feature where this information can be logged (the user failed to authenticate, we know the topic name, we can log a failure cause and reply with 403). Here, at the authentication phase it may be a problem. Undertow `AuthenticationMechanism` implementations that users can make use of or the `IdentityManager` interface are not aware of topics.